[ADD] events: registration desk doc (17) (#9348)
odoo/documentation: content/applications/marketing/events/registration_desk.rst

=================
Registration desk
=================

When an event is finally happening, it's important to grant access to people who registered to
attend it. Not only does that allow deserving people to enter the event, it also adds valuable data

> to the Odoo *Events* reporting metrics, which can be used for in-depth analysis in the future.

Reviewer: Also, this doc doesn't mention the adding data for reporting later. Will this be in a separate doc? If not, I think it'd be good to mention very briefly at the end of the doc how the data is stored in the report!

> to the Odoo *Events* reporting metrics, which can be used for in-depth analysis in the future.

Author: I believe I *do* mention in this intro (and in your suggested concise version) the adding of attendee data in reporting metrics for later analysis. I added a more specific wording in my revision. And yes, I plan on creating a reporting-specific doc(s) for this app in the very near future. Then, I can come back and add a reference link to it. 👍

   :align: center
   :alt: The Attendees page, via the Registration Desk, located in the Odoo Events application.

.. note::
   The :guilabel:`Attendees` page can also be viewed as a :icon:`oi-view-list` :guilabel:`List`,
   :icon:`fa-calendar` :guilabel:`Calendar`, or :icon:`fa-area-chart` :guilabel:`Graph`. Each view
   option is accessible, via their respective icons, located in the upper-right corner of the
   :guilabel:`Attendees` page.

On the :guilabel:`Attendees` page, in the default :icon:`oi-view-kanban` :guilabel:`Kanban` view,
each attendee card displays that person's name, which event they are registered to attend, their
associated company (if applicable), what ticket tier they purchased (if applicable), along with two
buttons: a :icon:`fa-check` :guilabel:`(checkmark)` and :icon:`fa-undo`
:guilabel:`(counter-clockwise arrow)`.

To grant access to a person, and officially mark them as attended, click the :icon:`fa-check`
:guilabel:`(checkmark)` button on that specific attendee's card.

If a mistake was made, or if the incorrect attendee was marked as attended, click the
:icon:`fa-undo` :guilabel:`(counter-clockwise arrow)` button on that specific attendee's card.

> It is recommended to use an event-specific filter on the :guilabel:`Attendees` page, via the
> :guilabel:`Search...` bar, to avoid having to search through a master list of attendees related to
> every event in the database.

Reviewer: ```suggestion
Use an event-specific filter on the :guilabel:`Attendees` page, via the
:guilabel:`Search...` bar, to avoid searching through a master list of all event attendees.
```

Author: In my opinion, posing this as a recommendation seems more appropriate, as it is *technically* not necessary for them to complete the task. But I fully understand (and appreciate) the suggestion to make things a bit more concise.

A simple way to do that is to click the :icon:`fa-sort-desc` :guilabel:`(downward arrow)` beside the
:guilabel:`Search...` bar to reveal a drop-down menu with :guilabel:`Filters`, :guilabel:`Group By`,
and :guilabel:`Favorites` options.

> 

Reviewer: Can link to the filters doc in a seealso block (:

https://www.odoo.com/documentation/17.0/applications/essentials/search.html

A quick way to organize the attendees on the :guilabel:`Attendees` page is to click

> :guilabel:`Event` in the :guilabel:`Group By` column. Then, click away to remove the drop-down menu

Reviewer: to further specify that we're clicking the option in the filters/groupby/fav drop-down
```suggestion
the :guilabel:`Event` option in the :guilabel:`Group By` column. Then, click away to remove the drop-down menu
```

=================
Registration desk
=================

Use the *Registration Desk* feature in Odoo *Events* to grant event access to registered attendees
as they arrive, and store valuable attendee-related data in the reporting metrics for future
analysis.

Registration Desk page
======================

First, the person (or people) in charge of granting access to event attendees, would ideally have a
compatible mobile-device open to their Odoo database.

.. note::
   The *Registration Desk* feature works with laptops that have cameras, but mobile devices are more
   convenient for managing attendee access.

> Then, open the :menuselection:`Events app`, and click the :guilabel:`Registration Desk` header menu

Reviewer 2: Follow up from previous comment (remove 6 lines and replace with half a line)

```suggestion
On a mobile device (on the Odoo app or in a browser), open the :menuselection:`Events app`, and click the :guilabel:`Registration Desk` header menu
```

option. Doing so, reveals a separate :guilabel:`Registration Desk` page.

.. image:: registration_desk/registration-desk-page.png
   :align: center
   :alt: The Registration Desk page in the Odoo Events application.

There is only a :guilabel:`Registration Desk` square in the center of the page, with a simple
message, reading: :guilabel:`Welcome to (Company Name)`. Beneath that, the company logo that has

> been uploaded in the database appears.

Reviewer 2: Not sure if necessary? Or can add this info as a more concise caption to the screenshot.

click the :guilabel:`Scan a badge` option.

Upon doing so, Odoo requests permission to access the camera connected to the device currently being
used. Odoo **must** be granted access to the camera, in order for the :guilabel:`Scan a badge`
option to work.

Once Odoo has access to the camera, and :guilabel:`Scan a badge` is clicked, a :guilabel:`Barcode
Scanner` pop-up window, showing the camera's point-of-view. There is also a specified view finder
box present, which can be enlarged or shrunk, accordingly, using the :icon:`fa-crop`
:guilabel:`(crop)` icon.

.. image:: registration_desk/barcode-scanner-window.png
   :align: center
   :alt: The Barcode Scanner window of the Registration Desk in the Odoo Events application.

Once the badge code is in the middle of the view finder window, the code is scanned, the
:guilabel:`Barcode Scanner` pop-up window disappears, and the attendee is granted access to the
event. Once the code is scanned, their attendance is logged in the Odoo *Events* app.

If the barcode being scanned is invalid, an error pop-up message appears in the upper-right corner.

Select attendee
===============

> If users wish to manually grant access to event attendees as they show up to the event, they would
> navigate to the :guilabel:`Registration Desk` page (:menuselection:`Events app --> Registration
> Desk`), and click the :guilabel:`Select Attendee` option.

Reviewer 2: Doc wide: can be more concise with instruction by removing conditional or passive language (e.g. users, "if" statements, etc).

```suggestion
Manually grant access to event attendees by navigating to :menuselection:`Events app -->
Registration Desk` and selecting the :guilabel:`Select Attendee` option.
```

Upon doing so, Odoo reveals a separate :guilabel:`Attendees` page, showcasing all the attendees for
every event in the database, in a default :icon:`oi-view-kanban` :guilabel:`Kanban` view.

.. image:: registration_desk/attendees-page.png
   :align: center
   :alt: The Attendees page, via the Registration Desk, located in the Odoo Events application.

.. note::
   The :guilabel:`Attendees` page can also be viewed as a :icon:`oi-view-list` :guilabel:`List`,
   :icon:`fa-calendar` :guilabel:`Calendar`, or :icon:`fa-area-chart` :guilabel:`Graph`. Each view
   option is accessible, via their respective icons, located in the upper-right corner of the

>    :guilabel:`Attendees` page.

Reviewer 2: Unnecessary unless formatted as a **Tip** to a preferred view that our internal team prefers (like **List** view).

On the :guilabel:`Attendees` page, in the default :icon:`oi-view-kanban` :guilabel:`Kanban` view,
each attendee card displays that person's name, which event they are registered to attend, their
associated company (if applicable), what ticket tier they purchased (if applicable), along with two
buttons: a :icon:`fa-check` :guilabel:`(checkmark)` and :icon:`fa-undo`
:guilabel:`(counter-clockwise arrow)`.

To grant access to a person, and officially mark them as attended, click the :icon:`fa-check`
:guilabel:`(checkmark)` button on that specific attendee's card.

If a mistake was made, or if the incorrect attendee was marked as attended, click the
:icon:`fa-undo` :guilabel:`(counter-clockwise arrow)` button on that specific attendee's card.

> It is recommended to use an event-specific filter on the :guilabel:`Attendees` page, via the

Reviewer 2: would format as a **tip** and give an example of "event-specific filter"
